include/psa/crypto.h

```diff
@@ -4045,6 +4045,597 @@ psa_status_t psa_generate_key(const psa_key_attributes_t *attributes,
+ *                              time. The only guarantee is that lower values
+ *                              for \p max_ops means functions will block for a
+ *                              lesser maximum amount of time and conversely
+ *                              larger values will mean blocking for a larger
+ *                              maximum amount of time. The functions
+ *                              \c psa_sign_interruptible_get_num_ops() and
+ *                              \c psa_verify_interruptible_get_num_ops() are
+ *                              provided to help with tuning this value.
+ *
+ * \note                        Although the interpretation of this value is
+ *                              implicitly implementation defined, this value
+ *                              needed a default, thus this value defaults to
+ *                              #PSA_INTERRUPTIBLE_MAX_OPS_UNLIMITED, which
+ *                              means the whole operation will be done in one
+ *                              go, regardless of the number of ops required.
+ *
+ * \note                        If more ops are needed to complete a
+ *                              computation, #PSA_OPERATION_INCOMPLETE will be
+ *                              returned by the function performing the
+ *                              computation. It is then the caller's
+ *                              responsibility to either call again with the
+ *                              same parameters until it returns 0 or an error
+ *                              code; or to free the relevant context if the
+ *                              operation is to be aborted.
+ *
+ * \note                        The interpretation of this maximum number is
+ *                              obviously also implementation defined. On a
+ *                              hard real time system, this can indicate a hard
+ *                              deadline, which is good, as a real-time system
+ *                              needs a guarantee of not spending more than X
+ *                              time, however care must be taken to avoid the
+ *                              situation whereby calls just return, not being
+ *                              able to do any actual work within the allotted
+ *                              time.  On a non-real-time system, the
+ *                              implementation can be more relaxed, but again
+ *                              whether this number should be interpreted as as
+ *                              hard or soft limit or even whether a less than
+ *                              or equals as regards to ops executed in a
+ *                              single call is implementation defined.
+ *
+ * \warning                     With implementations that interpret this number
+ *                              as a hard limit, setting this number too small
```

Comment:
This also sounds like a potential for something pretty complex, which seems a bit worrying, given its not something we can test. The best I can think here is to have some kind of (driver wrapped) call which takes the initialised operation as an input, and returns the min max_ops value. The internal implementation is always going to return 1, however. Does this sound acceptable?

Reply:
I agree that it sounds complex, especially with drivers. I'd like to avoid this complexity in the first iteration, so my preferred outcome for this comment is:

* You do nothing in this pull request.
* I file this as an API improvement to be done before the API is released, and we think about it calmly.